### PR TITLE
Add helpers for decoding content from requests with a body

### PR DESCRIPTION
**Pull request by Joannis**

### CODE REVIEW

File: Sources/Vapor/Engine/Responder.swift [hunk 1]
```diff
@@ -28,6 +28,30 @@ public struct RouteResponder<T>: Responder
     }
 }
 
+/// A basic, closure-based responder.
+public struct ContentRouteResponder<C, T>: Responder
+    where C: Content, T: ResponseEncodable
```
Comment (tanner0101) [inline Sources/Vapor/Engine/Responder.swift]: this should be just `RequestDecodable`. no reason to limit to `Content`.

File: Sources/Vapor/Routing/Router+Method.swift [hunk 2]
```diff
@@ -17,6 +17,22 @@ extension Router {
         self.register(route: route)
         return route
     }
+    
+    /// Registers a route handler at the supplied path.
+    @discardableResult
+    fileprivate func on<C, T>(
+        _ method: HTTPMethod,
+        to path: [PathComponent],
```
Comment (tanner0101) [inline Sources/Vapor/Routing/Router+Method.swift]: inconsistent labels (`to:` vs `path:` on the other calls). IMO this should be `at:` to match what we do in other parts of the framework, like in CodableKit

File: Sources/Vapor/Engine/Responder.swift [hunk 3]
```diff
@@ -28,6 +28,30 @@ public struct RouteResponder<T>: Responder
     }
 }
 
+/// A basic, closure-based responder.
+public struct ContentRouteResponder<C, T>: Responder
```
Comment (tanner0101) [inline Sources/Vapor/Engine/Responder.swift]: this should probably be `RequestDecodableResponder` as well

File: Sources/Vapor/Routing/Router+Method.swift (hunk 2 above)
Comment (Joannis) [reply Sources/Vapor/Routing/Router+Method.swift]: That was just mimicking the existing function signature above. I'll have to change the existing public function, too, then.

**Comment by twof** [comment]
I really really like this one

**Pull request merged**